packages/select/src/components/omnibar/omnibar.tsx

```diff
@@ -90,31 +73,18 @@ export class Omnibar<T> extends React.PureComponent<IOmnibarProps<T>, IOmnibarSt
         return (
             <this.TypedQueryList
                 {...restProps}
-                activeItem={this.state.activeItem}
                 initialContent={initialContent}
-                onActiveItemChange={this.handleActiveItemChange}
                 onItemSelect={this.props.onItemSelect}
-                query={this.state.query}
                 ref={this.refHandlers.queryList}
                 renderer={this.renderQueryList}
             />
         );
     }
 
-    public componentWillReceiveProps(nextProps: IOmnibarProps<T>) {
-        const { isOpen } = nextProps;
-        const canClearQuery = !this.props.isOpen && isOpen && this.props.resetOnSelect;
-
-        this.setState({
-            activeItem: canClearQuery ? this.props.items[0] : this.state.activeItem,
-            query: canClearQuery ? "" : this.state.query,
-        });
-    }
-
     private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
         const { inputProps = {}, isOpen, overlayProps = {} } = this.props;
         const { handleKeyDown, handleKeyUp } = listProps;
-        const handlers = isOpen && !this.isQueryEmpty() ? { onKeyDown: handleKeyDown, onKeyUp: handleKeyUp } : {};
+        const handlers = isOpen && listProps.query.length > 0 ? { onKeyDown: handleKeyDown, onKeyUp: handleKeyUp } : {};
```

Comment:
query is optional on props, maybe check null

Reply:
`listProps.query` is always defined as it comes from QueryList